### DISCUSSION TITLE
Nullability flow analysis improvement

### DISCUSSION THREAD

**Discussion by TahirAhmadov** [dotnet/csharplang#5713] [Language Ideas] [answered]
In the below example, the `t.Trim()` shows a nullability warning, but it shouldn't - it's known that if `s` is not null, then `t` cannot be null.
```cs
void Foo(string? s, string? t)
{
  if (s != null && t == null) return;
  if (s != null)
    Console.Write(t.Trim());
}
```

**Comment by huoyaoyuan** [comment, marked as answer]
The nullable flow analysis doesn't and can't track expression combinations, because the complexity is infinite. Refactor to nested if, or just use `!`.
